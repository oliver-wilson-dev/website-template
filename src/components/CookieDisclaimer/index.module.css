.container {
    position: fixed;
    bottom: -200px;
    left: 0;
    right: 0;
    z-index: 2;
    display: flex;
    justify-content: center;
    align-items: center;
    transition: bottom 0.5s;
    transition-timing-function: cubic-bezier(0.68, -0.55, 0.265, 1.55);

    &.loaded {
        bottom: 0;
    }

    &.closed {
        bottom: -200px;
    }
}

.positionWrapper {
    background: #F4D03F;
    border-radius: 40px;
    margin: 30px 10px;
    padding: 20px;
    max-width: 400px;

    display: flex;
    justify-content: space-evenly;
    align-items: center;
}

.text {
    color: black;
    font-size: 14px;
}

.message {
    flex-basis: 80%;
}

.notificationEmoji {
    width: 30px;
    height: 30px;
}

.tick {
    width: 40px;
    height: 35px;
    border: none;
    stroke: black;
    stroke-width: 1;
}

.notificationElement {
    margin: 0 5px;
}

.bannerContent {
    margin: 0 10px;
    display: flex;
    flex-direction: column;
}

.acceptButton {
    background: black;
    border: none;
    text-transform: uppercase;
    border: 1px solid;
    width: 50%;
    margin: 0 auto;
}

.learnMoreBtn {
    text-decoration: underline;
    background: transparent;
    border: none;
    padding: 0;
}