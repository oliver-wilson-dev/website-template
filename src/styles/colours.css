$background-grey: #e9ecef;
$dark-grey:  #333333;
$text-dark-grey:  #333333;
$text-light-grey:  #ffffff;
