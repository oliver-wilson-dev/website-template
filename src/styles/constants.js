export const LIGHT_THEME = Symbol('LIGHT_THEME');
export const DARK_THEME = Symbol('DARK_THEME');
