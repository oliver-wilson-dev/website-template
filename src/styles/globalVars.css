$max-content-width: 1000px;
$navigation-height: 50px;