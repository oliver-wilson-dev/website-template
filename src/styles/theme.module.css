@import "./colours.css";

.light {
    --background-colour: $background-grey;
    --background-colour-inverse: $dark-grey;
    --text-colour: $text-dark-grey;
    --text-colour-inverse: $text-light-grey;
    --side-menu-focus: $dark-grey;
}

.dark {
    --background-colour: $dark-grey;
    --background-colour-inverse: $background-grey;
    --text-colour: $text-light-grey;
    --text-colour-inverse: $text-dark-grey;
    --side-menu-focus: $background-grey;
}
