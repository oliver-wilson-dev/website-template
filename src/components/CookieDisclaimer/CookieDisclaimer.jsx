import React, { useState, useEffect } from 'react';
import cn from 'classnames';
import styles from './index.module.css';
import NotificationEmoji from './assets/notification.svg';
import TickEmoji from './assets/tick.svg';

const CookieDisclaimerMessage = ({ message, cookiesAccepted }) => {
  const [acknowledged, setAcknowledged] = useState(false);
  const [loaded, setLoaded] = useState(false);
  const [closed, setClosed] = useState(false);

  useEffect(() => {
    setLoaded(true);
  }, []);

  const onTransitionEnd = () => {
    if (acknowledged) {
      setClosed(true);
    }
  };

  const acknowledgeBanner = () => {
    setAcknowledged(true);
    cookiesAccepted();
  };

  return !closed && (
    <div
      className={cn(styles.container, {
        [styles.loaded]: loaded && !acknowledged,
        [styles.closed]: acknowledged
      })}
      onTransitionEnd={onTransitionEnd}
    >
      <div className={styles.positionWrapper}>
        <NotificationEmoji className={cn(
          styles.notificationEmoji,
          styles.notificationElement
        )}
        />
        <span className={cn(
          styles.message,
          styles.text,
          styles.notificationElement,
        )}
        >
          {message}
        </span>
        <TickEmoji
          className={cn(styles.tick, styles.notificationElement)}
          onClick={acknowledgeBanner}
        />
      </div>
    </div>
  );
};

export default CookieDisclaimerMessage;
