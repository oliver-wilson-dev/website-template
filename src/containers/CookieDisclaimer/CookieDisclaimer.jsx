import { connect } from 'react-redux';
import CookieDisclaimer from '../../components/CookieDisclaimer';
import { getMessage } from '../../state/selectors';
import { cookiesAccepted } from '../../state/actions';

const mapStateToProps = (state) => ({
  message: getMessage({ state }),
});

const mapDispatchToProps = (dispatch) => ({
  cookiesAccepted: () => dispatch(cookiesAccepted())
});

export default connect(mapStateToProps, mapDispatchToProps)(CookieDisclaimer);
