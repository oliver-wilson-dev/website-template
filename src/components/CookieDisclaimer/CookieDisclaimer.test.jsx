import React from 'react';
import { shallow, mount } from 'enzyme';
import { act } from 'react-dom/test-utils';
import styles from './index.module.css';

import CookieDisclaimer from '.';

jest.mock('./assets/tick.svg', () => {
  const Tick = () => null;

  return Tick;
});

jest.mock('./assets/notification.svg', () => {
  const Notification = () => null;

  return Notification;
});

const defaultProps = {
  cookiesAccepted: jest.fn(),
  message: 'hello world'
};

const render = (renderMethod = shallow, props = {}) => renderMethod(
  <CookieDisclaimer
    {...defaultProps}
    {...props}
  />
);

describe('Cookie Disclaimer component', () => {
  describe('when first loaded', () => {
    it('should render correctly', () => {
      expect(render()).toMatchSnapshot();
    });

    it('should then apply the loaded class', () => {
      expect(render(mount)).toMatchSnapshot();
    });

    describe('and the initial animation happens', () => {
      it('should not close', () => {
        const component = render(mount);
        const container = component.find(`.${styles.container}`);
        container.props().onTransitionEnd();

        expect(container.hasClass(styles.loaded)).toBe(true);
        expect(container.hasClass(styles.closed)).toBe(false);
      });
    });

    describe('when the user accepts the policy', () => {
      it('should render correctly', () => {
        const component = render();

        component.find(`.${styles.tick}`).simulate('click');

        expect(component).toMatchSnapshot();
      });

      it('should call cookiesAccepted', () => {
        const component = render();

        component.find(`.${styles.tick}`).simulate('click');

        expect(defaultProps.cookiesAccepted).toHaveBeenCalled();
      });

      describe('after the transition ends', () => {
        it('should render null', () => {
          const component = render();

          component.find(`.${styles.tick}`).simulate('click');

          act(() => {
            component.find(`.${styles.closed}`).props().onTransitionEnd();
          });

          component.update();

          expect(component).toMatchSnapshot();
        });
      });
    });
  });
});
