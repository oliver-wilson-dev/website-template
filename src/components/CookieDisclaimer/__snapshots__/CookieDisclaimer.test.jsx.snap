// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cookie Disclaimer component when first loaded should render correctly 1`] = `
<div
  className="container"
  onTransitionEnd={[Function]}
>
  <div
    className="positionWrapper"
  >
    <Notification
      className="notificationEmoji notificationElement"
    />
    <span
      className="message text notificationElement"
    >
      hello world
    </span>
    <Tick
      className="tick notificationElement"
      onClick={[Function]}
    />
  </div>
</div>
`;

exports[`Cookie Disclaimer component when first loaded should then apply the loaded class 1`] = `
<CookieDisclaimerMessage
  cookiesAccepted={[MockFunction]}
  message="hello world"
>
  <div
    className="container loaded"
    onTransitionEnd={[Function]}
  >
    <div
      className="positionWrapper"
    >
      <Notification
        className="notificationEmoji notificationElement"
      />
      <span
        className="message text notificationElement"
      >
        hello world
      </span>
      <Tick
        className="tick notificationElement"
        onClick={[Function]}
      />
    </div>
  </div>
</CookieDisclaimerMessage>
`;

exports[`Cookie Disclaimer component when first loaded when the user accepts the policy after the transition ends should render null 1`] = `""`;

exports[`Cookie Disclaimer component when first loaded when the user accepts the policy should render correctly 1`] = `
<div
  className="container closed"
  onTransitionEnd={[Function]}
>
  <div
    className="positionWrapper"
  >
    <Notification
      className="notificationEmoji notificationElement"
    />
    <span
      className="message text notificationElement"
    >
      hello world
    </span>
    <Tick
      className="tick notificationElement"
      onClick={[Function]}
    />
  </div>
</div>
`;
